src/material.h

```diff
@@ -75,4 +77,13 @@ class material_type
         const mat_burn_data &burn_data( size_t intensity ) const;
 };
 
+namespace materials
```

Comment:
> That isn't an answer to YAGNI

Isn't it?

> Your breaking consistency with existing code

I won't argue with that, just quote a nipped of it:
```
    mission_type::reset();
    item_controller->reset();
    mutations_category.clear();
    reset_bionics();
    MonsterGenerator::generator().reset();
    clear_techniques_and_martial_arts();
```
> and need a reason for this.

I have written my reasons above. Let's put it also this way: `generic_factory` expects its items to have `load()` and `check()` member functions. Of course, I can write `check()` and `check_all()` functions inside one class, but: 
1. they're clearly on different levels of abstraction
2. the names are misleading. I have to remember that `check()` != `check_all()` and their special meaning.